### PR TITLE
[#2366] Make arrows smaller and move them around

### PR DESCRIPTION
**Pull request by punchagan**
To avoid overlaps, the arrows are placed at slightly different positions
based on the index of the connector for which we are using an arrow, and
the curviness of the connectors is also changed.

| Before        | After         |
| ------------- |---------------|
| mon oct 3 16 58 30 ist 2016      | mon oct 3 16 57 39 ist 2016  |

### CODE REVIEW

**Comment by punchagan** [comment]
This doesn't completely fix the problems with #2366, but mitigates them.  We could also add different colors for the connectors+arrows to make it clearer.  

But, to have curves which are not overlapping at all would be a lot more work.

File: akvo/rsr/static/scripts-src/project-hierarchy.js [hunk 1]
```diff
@@ -2,23 +2,29 @@ jsPlumb.ready(function() {
...
-            two_arrows = [
-                [ "Arrow", { location:0.7 }, arrowCommon ],
-                [ "Arrow", { location:0.3, direction:-1 }, arrowCommon ]
             ];
 
+        // Function to off-set arrows based on the index of connector on the
+        // element for which we are using the arrows
+        var get_two_arrows = function (index) {
+            var two_arrows = [
+                [ "Arrow", { location:(0.9 - index*0.1) }, arrowCommon ],
```
Comment (punchagan) [inline akvo/rsr/static/scripts-src/project-hierarchy.js]: min(0.5, v) needed here

**Comment by punchagan** [comment]
tue oct 4 12 55 08 ist 2016

This is one of the outputs I get by using entirely random colors.  We could be more methodical in choosing colors to make it look better, I guess. 

@loicsans might be able to give us more insights into this.

**Comment by loicsans** [comment]
@punchagan, as far as I can see we have 2 size arrows.
One better solution would be to choose one colour for the short arrow and an other one for the long one.

**Comment by punchagan** [comment]
tue oct 4 15 32 35 ist 2016

@loicsans We can have arrows of more than 2 sizes, actually, depending on how many siblings/children a project has.  But, I've implemented code that arrows of similar "size" are of the same color. ["size" is in quotes because, the library doesn't really let us detect size - the code indirectly interprets this, based on the index of the project in a list of sibling/child projects]

The colors I chose currently are random.  Since RSR doesn't have a color palette, etc. I'm not sure what colors to choose.  Any recommendations more than welcome!

**Comment by zzgvh** [comment]
When I try this on project 3094, which has a ton of children, some of the lines only show when you hover over them:
rsr arrows
I'm seeing this in FF and Safari.

**Comment by nadiagorchakova** [comment]
In the end, we decided not to implement different colours as it adds unnecessary complexity. Hierarchy for project 3094 looks funny no matter the colours though.

**Comment by punchagan** [comment]
@zzgvh Thanks for trying this on that project.  Can you tell me how to import that data locally? 
It's because of the `curviness` being set to crazy numbers for the Bezier curves, I think. I'll look into this, tomorrow.

**Comment by punchagan** [comment]
This is what we can get in the "small" case. 
thu oct 6 16 55 20 ist 2016

And this is what we get for the "large" case. 
thu oct 6 16 49 57 ist 2016

Getting anything better than this will be significantly more work, either with this or another library. (estimate: somewhere between 16 and 32). 

If the current results seem better than the older ones (which they do, to me) we can merge it, otherwise we can just close this PR.

**Pull request merged**